ml-agents/mlagents/trainers/trainer.py

```diff
@@ -20,22 +21,33 @@ class UnityTrainerException(UnityException):
 class Trainer(object):
     """This class is the base class for the mlagents.trainers"""
 
-    def __init__(self, brain, trainer_parameters, training, run_id):
+    def __init__(self, brain, trainer_parameters, training, run_id,
+                 debug_flag):
         """
         Responsible for collecting experiences and training a neural network model.
         :BrainParameters brain: Brain to be trained.
         :dict trainer_parameters: The parameters for the trainer (dictionary).
         :bool training: Whether the trainer is set for training.
         :int run_id: The identifier of the current run
+        :bool debug_flag: Checks debug_flag
         """
         self.param_keys = []
         self.brain_name = brain.brain_name
         self.run_id = run_id
         self.trainer_parameters = trainer_parameters
+        self.time_start_experience_collection = None
+        self.delta_last_experience_collection = None
+        self.cumulative_returns_since_policy_update = []
+        self.last_mean_return = None
+        self.last_buffer_length = 0
         self.is_training = training
         self.stats = {}
         self.summary_writer = None
+        self.training_metrics  = []
         self.policy = None
+        if debug_flag:
+            logger.setLevel('DEBUG')
```

Comment:
Seems like an alternative to setting the logging level in multiple places would be to set the logging level at the entrypoint and have all of these modules use shared loggers (`trainer_logger`, `env_logger`, etc).

Reply:
Good idea, will fix

Reply:
PTAL